crates/bevy_reflect/src/map.rs

```diff
@@ -197,11 +197,12 @@ impl MapInfo {
 #[macro_export]
 macro_rules! hash_error {
     ( $key:expr ) => {{
-        let type_name = match (*$key).get_represented_type_info() {
-            None => "Unknown",
-            Some(s) => s.type_path(),
-        };
-        format!("the given key {} does not support hashing", type_name).as_str()
+        let type_path = (*$key).reflect_type_path();
```

Comment:
This should be okay, but I'm wondering if maybe we should split the output based on `Reflect::is_dynamic`.

Like, for non-dynamic types we do exactly what you have here, but for dynamic types we can use `get_represented_type_info` like you had before to do something like:

```rust
let type_path = (*$key).reflect_type_path();
// Handle the `Reflect::is_dynamic == true` case:
let output = match (*$key).get_represented_type_info() {
  // Handle dynamic types that do not represent a type (i.e a plain `DynamicStruct`):
  None => format!("the dynamic type `{}` does not support hashing", type_path),
  // Handle dynamic types that do represent a type (i.e. a `DynamicStruct` proxying `Foo`):
  Some(s) => format!(
    "the dynamic type `{}` (representing `{}`) does not support hashing",
    type_path,
    s.type_path()
  ),
};
```

This should hopefully give enough detail so that users can find where they may accidentally be converting their types to a dynamic type.

Reply:
@MrGVSV  youre probably right. im just scared that it will get too complex for its own good, when the simple solution already works great. Ill try adding it and update the pull request, and you decide which one is better